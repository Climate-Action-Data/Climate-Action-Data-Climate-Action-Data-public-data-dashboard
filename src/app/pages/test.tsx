'use client'
import { type NextPage } from "next";

const Test: NextPage = () => {
    return (
        <div>
            <h1>Hello</h1>
            <p>data</p>
        </div>
    )
}

export default Test