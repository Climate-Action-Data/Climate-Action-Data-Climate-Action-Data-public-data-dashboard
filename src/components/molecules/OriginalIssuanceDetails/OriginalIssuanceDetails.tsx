import { Link, SimpleGrid, Skeleton } from '@chakra-ui/react'
import { useTranslation } from 'react-i18next'

import { DetailWidget } from '@/components/atoms/DetailWidget/DetailWidget'
import { Unit } from '@/@types/Unit'
import { generateProjectUrl } from '@/utils/RequestHelpers'
import { SearchFlow } from '@/@types/Search'

import { formatDate } from '@/utils/DateFormat'
import { DateFormats } from '@/@types/DateFormats'

interface OriginalIssuanceDetailsProps {
  unit?: Unit
}

const DEFAULT_COLUMN_MOBILE = 2
const DEFAULT_COLUMN_WEB = 3

export const OriginalIssuanceDetails = (props: OriginalIssuanceDetailsProps) => {
  const { unit } = props
  const { t } = useTranslation(`unitDetails`)
  const { t: tHome } = useTranslation(`home`)
  if (!unit) {
    return <Skeleton height="100px" />
  }

  const renderMonitoringDate = (monitoringPeriodStart: string, monitoringPeriodEnd: string) => {
    if (monitoringPeriodStart && monitoringPeriodEnd) {
      return `${formatDate(monitoringPeriodStart, DateFormats.YYYY_MM_DD)} - ${formatDate(monitoringPeriodEnd, DateFormats.YYYY_MM_DD)}`
    } else {
      return tHome(`noData`)
    }
  }

  return (
    <SimpleGrid columns={[DEFAULT_COLUMN_MOBILE, DEFAULT_COLUMN_WEB]} gap="24px">
      <DetailWidget title={t(`originalIssuance.projectIssuedTo`)}>
        <Link href={`${generateProjectUrl(unit.project?.warehouseProjectId, SearchFlow.UNIT)}`} textDecoration="underline">
          {unit?.project?.name ?? tHome(`noData`)}
        </Link>
      </DetailWidget>
      <DetailWidget title={t(`originalIssuance.issuanceBatchSerialNumber`)}>{unit?.issuance?.issuanceBatchSerial ?? tHome(`noData`)}</DetailWidget>
      <DetailWidget title={t(`originalIssuance.vintage`)}>{unit?.vintage ?? tHome(`noData`)}</DetailWidget>
      <DetailWidget title={t(`originalIssuance.quantityIssued`)}>{unit?.credits ?? tHome(`noData`)}</DetailWidget>
      <DetailWidget title={t(`originalIssuance.unitType`)}>{unit?.type ?? tHome(`noData`)}</DetailWidget>
      <DetailWidget title={t(`originalIssuance.monitoringPeriod`)}>{renderMonitoringDate(unit?.monitoringPeriodStart, unit?.monitoringPeriodEnd)}</DetailWidget>
    </SimpleGrid>
  )
}
