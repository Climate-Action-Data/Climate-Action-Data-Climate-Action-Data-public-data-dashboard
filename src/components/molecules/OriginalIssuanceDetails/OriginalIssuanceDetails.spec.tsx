import { render } from '@testing-library/react'
import { OriginalIssuanceDetails } from './OriginalIssuanceDetails'
import { UNIT_DETAIL } from '@/test/TestOvermindMockData'

describe(`RetirementDetails`, () => {
  it(`renders correctly with no project`, () => {
    const { container } = render(<OriginalIssuanceDetails unit={undefined} />)
    expect(container).toMatchSnapshot()
  })

  it(`renders correctly with validation`, () => {
    const { container } = render(<OriginalIssuanceDetails unit={UNIT_DETAIL} />)
    expect(container).toMatchSnapshot()
  })
})
