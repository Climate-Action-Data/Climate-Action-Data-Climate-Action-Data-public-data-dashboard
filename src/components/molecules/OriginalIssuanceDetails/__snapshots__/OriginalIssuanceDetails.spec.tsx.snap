// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetirementDetails renders correctly with missing details 1`] = `
<div>
  <div
    class="css-1h55qqd"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Project Issued To
      </p>
      <p
        class="chakra-text css-0"
      >
        <a
          class="chakra-link css-1081t4c"
          href="/project?id=a442a765-50f4-43dc-aa38-24e08d88c821"
        >
          --
        </a>
      </p>
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Issuance Batch Serial N0.
      </p>
      <p
        class="chakra-text css-0"
      >
        --
      </p>
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Vintage
      </p>
      <p
        class="chakra-text css-0"
      >
        --
      </p>
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Quantity Issued
      </p>
      <p
        class="chakra-text css-0"
      >
        --
      </p>
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Unit Type
      </p>
      <p
        class="chakra-text css-0"
      >
        --
      </p>
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Monitoring Period
      </p>
      <p
        class="chakra-text css-0"
      >
        --
      </p>
    </div>
  </div>
</div>
`;

exports[`RetirementDetails renders correctly with no project 1`] = `
<div>
  <div
    class="chakra-skeleton css-1p3r0co"
  />
</div>
`;

exports[`RetirementDetails renders correctly with validation 1`] = `
<div>
  <div
    class="css-1h55qqd"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Project Issued To
      </p>
      <p
        class="chakra-text css-0"
      >
        <a
          class="chakra-link css-1081t4c"
          href="/project?id=a442a765-50f4-43dc-aa38-24e08d88c821"
        >
          Proyecto de Mitigaci√≥n Forestal Bonanza Verde
        </a>
      </p>
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Issuance Batch Serial N0.
      </p>
      <p
        class="chakra-text css-0"
      >
        136cadbc-83a5-4c43-a172-baf7770796fc
      </p>
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Vintage
      </p>
      <p
        class="chakra-text css-0"
      >
        2019
      </p>
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Quantity Issued
      </p>
      <p
        class="chakra-text css-0"
      >
        1102
      </p>
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Unit Type
      </p>
      <p
        class="chakra-text css-0"
      >
        Reduction - nature
      </p>
    </div>
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1xi4xed"
      >
        Monitoring Period
      </p>
      <p
        class="chakra-text css-0"
      >
        2019-12-16T02:50:47.000Z
      </p>
    </div>
  </div>
</div>
`;
